fix(integration): listTags filter by parentId returns empty for newly-created child

## Context

`listTags({ parentId })` returns `[]` when called immediately after creating a child tag whose parent is `parentId`. The expected child tag should appear. This is one of the long-standing integration-suite failures present on every release tag since v1.0.0.

**Test:** `tests/contract/adapter.contract.ts:393` — `listTags filters by parentId (nested tag hierarchy)`. Reproducible locally.

```bash
OMNIFOCUS_INTEGRATION=1 pnpm vitest run \
  tests/integration/transportRouter.contract.test.ts \
  -t "listTags filters by parentId"
# → AssertionError: expected [] to deeply equal [ 'eHI7L46obdg' ]
```

Tracks: cluster of integration test failures repro'd in [run 25129921989](https://github.com/torsday/omnifocus-mcp/actions/runs/25129921989).

## Acceptance Criteria

- [ ] After `createTag({ name: "child", parentId: top })`, calling `listTags({ parentId: top })` returns an array containing the new child's id
- [ ] `tests/integration/transportRouter.contract.test.ts > tags — CRUD + filter > listTags filters by parentId (nested tag hierarchy)` passes locally and on the `macos-omnifocus` runner
- [ ] Root cause noted in commit message — distinguish a real listTags bug (filter not propagated) from a sync-visibility issue (newly-created tag not yet visible)

## Technical Notes

- Suspects, in order of likelihood: (1) `scripts/jxa/tag_list.js` ignores or drops the `parentId` filter argument; (2) `JxaTransport.listTags` doesn't pass `parentId` through to the script; (3) the OmniFocus tag database needs an explicit refresh / `flush` after `createTag` before the parent edge is queryable.
- Distinct from the broader `createTask projectId` cluster — that's task-side, this is tag-side. They likely share a flavor (parent-edge not assigned-or-not-readable) but probably need separate fixes.
- Quick test for the sync-visibility hypothesis: insert a 100ms sleep between `createTag` and `listTags` in the test and see if the child appears. If yes → script-flush bug. If no → real listTags filter bug.

## Dependencies

- Tracks: cluster of integration test failures repro'd in [run 25129921989](https://github.com/torsday/omnifocus-mcp/actions/runs/25129921989)
- Independent of the `createTask projectId` cluster

--- a/src/scripts/jxa/forecast_get.js
+++ b/src/scripts/jxa/forecast_get.js
@@ -73,7 +73,9 @@ function run(argv) {
     try {
       const tags = task.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/perspective_evaluate.js
+++ b/src/scripts/jxa/perspective_evaluate.js
@@ -61,7 +61,9 @@ function run(argv) {
       try {
         const tags = task.tags();
         for (let i = 0; i < tags.length; i++) {
-          tagIds.push(tags[i].id());
+          try {
+            tagIds.push(tags[i].id());
+          } catch (_tagErr) {}
         }
       } catch (_e) {}
 

--- a/src/scripts/jxa/project_create.js
+++ b/src/scripts/jxa/project_create.js
@@ -36,7 +36,9 @@ function run(argv) {
     try {
       const tags = proj.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/project_get.js
+++ b/src/scripts/jxa/project_get.js
@@ -34,7 +34,9 @@ function run(argv) {
     try {
       const tags = proj.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/project_get_many.js
+++ b/src/scripts/jxa/project_get_many.js
@@ -32,7 +32,9 @@ function run(argv) {
     try {
       const tags = proj.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/project_list.js
+++ b/src/scripts/jxa/project_list.js
@@ -34,7 +34,9 @@ function run(argv) {
     try {
       const tags = proj.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/project_update.js
+++ b/src/scripts/jxa/project_update.js
@@ -37,7 +37,9 @@ function run(argv) {
     try {
       const tags = proj.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/tag_create.js
+++ b/src/scripts/jxa/tag_create.js
@@ -18,7 +18,20 @@ function run(argv) {
     let parentId = null;
     try {
       const p = tag.parent();
-      if (p && p.class() !== "document") parentId = p.id();
+      if (p) {
+        // OF 4.x: p.class() may throw — same defensive pattern as task_list.js #673.
+        let isDocument = false;
+        try {
+          isDocument = p.class() === "document";
+        } catch (_classErr) {}
+        if (!isDocument) {
+          try {
+            parentId = p.id();
+          } catch (_idErr) {
+            parentId = null;
+          }
+        }
+      }
     } catch (_e) {}
 
     let location = null;

--- a/src/scripts/jxa/tag_get.js
+++ b/src/scripts/jxa/tag_get.js
@@ -18,7 +18,21 @@ function run(argv) {
     let parentId = null;
     try {
       const p = tag.parent();
-      if (p && p.class() !== "document") parentId = p.id();
+      if (p) {
+        // OF 4.x: p.class() may throw on container objects (same pattern as
+        // containingProject().class() fix in task_list.js — see #673).
+        let isDocument = false;
+        try {
+          isDocument = p.class() === "document";
+        } catch (_classErr) {}
+        if (!isDocument) {
+          try {
+            parentId = p.id();
+          } catch (_idErr) {
+            parentId = null;
+          }
+        }
+      }
     } catch (_e) {}
 
     let location = null;

--- a/src/scripts/jxa/tag_get_many.js
+++ b/src/scripts/jxa/tag_get_many.js
@@ -18,7 +18,21 @@ function run(argv) {
     let parentId = null;
     try {
       const p = tag.parent();
-      if (p && p.class() !== "document") parentId = p.id();
+      if (p) {
+        // OF 4.x: p.class() may throw on container objects (same pattern as
+        // containingProject().class() fix in task_list.js — see #673).
+        let isDocument = false;
+        try {
+          isDocument = p.class() === "document";
+        } catch (_classErr) {}
+        if (!isDocument) {
+          try {
+            parentId = p.id();
+          } catch (_idErr) {
+            parentId = null;
+          }
+        }
+      }
     } catch (_e) {}
 
     let location = null;

--- a/src/scripts/jxa/tag_list.js
+++ b/src/scripts/jxa/tag_list.js
@@ -18,7 +18,24 @@ function run(argv) {
     let parentId = null;
     try {
       const p = tag.parent();
-      if (p && p.class() !== "document") parentId = p.id();
+      if (p) {
+        // OF 4.x: p.class() may throw on container objects (same pattern as
+        // containingProject().class() fix in task_list.js — see #673).
+        // When it throws we assume the parent is a real tag (not the document
+        // root) and fall through to p.id(); if p.id() also throws the outer
+        // catch leaves parentId = null, which is the safe default.
+        let isDocument = false;
+        try {
+          isDocument = p.class() === "document";
+        } catch (_classErr) {}
+        if (!isDocument) {
+          try {
+            parentId = p.id();
+          } catch (_idErr) {
+            parentId = null; // id() threw — treat as no parent (document root)
+          }
+        }
+      }
     } catch (_e) {}
 
     let location = null;

--- a/src/scripts/jxa/tag_update.js
+++ b/src/scripts/jxa/tag_update.js
@@ -18,7 +18,21 @@ function run(argv) {
     let parentId = null;
     try {
       const p = tag.parent();
-      if (p && p.class() !== "document") parentId = p.id();
+      if (p) {
+        // OF 4.x: p.class() may throw on container objects (same pattern as
+        // containingProject().class() fix in task_list.js — see #673).
+        let isDocument = false;
+        try {
+          isDocument = p.class() === "document";
+        } catch (_classErr) {}
+        if (!isDocument) {
+          try {
+            parentId = p.id();
+          } catch (_idErr) {
+            parentId = null;
+          }
+        }
+      }
     } catch (_e) {}
 
     let location = null;

--- a/src/scripts/jxa/task_create.js
+++ b/src/scripts/jxa/task_create.js
@@ -61,7 +61,9 @@ function run(argv) {
     try {
       const tags = task.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/task_get.js
+++ b/src/scripts/jxa/task_get.js
@@ -61,7 +61,9 @@ function run(argv) {
     try {
       const tags = task.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/task_get_many.js
+++ b/src/scripts/jxa/task_get_many.js
@@ -54,7 +54,9 @@ function run(argv) {
     try {
       const tags = task.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/task_list.js
+++ b/src/scripts/jxa/task_list.js
@@ -60,7 +60,12 @@ function run(argv) {
     try {
       const tags = task.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        // Guard per-element: a single bad tag object must not abort the loop
+        // and zero-out all tagIds, which would silently exclude this task
+        // from tagId-filter results (see #682).
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/task_search.js
+++ b/src/scripts/jxa/task_search.js
@@ -71,7 +71,9 @@ function run(argv) {
     try {
       const tags = task.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

--- a/src/scripts/jxa/task_update.js
+++ b/src/scripts/jxa/task_update.js
@@ -60,7 +60,9 @@ function run(argv) {
     try {
       const tags = task.tags();
       for (let i = 0; i < tags.length; i++) {
-        tagIds.push(tags[i].id());
+        try {
+          tagIds.push(tags[i].id());
+        } catch (_tagErr) {}
       }
     } catch (_e) {}
 

